sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -369,28 +373,120 @@ public static String stringJoin(String delimiter, List<String> values) {
+                return "";
+            case 1:
+                return values.get(0);
+            case 2:
+                return values.get(0) + delimiter + values.get(1);
+            case 3:
+                return values.get(0) + delimiter + values.get(1) + delimiter + values.get(2);
+            case 4:
+                return values.get(0) + delimiter + values.get(1) + delimiter + values.get(2) + delimiter
+                    + values.get(3);
+            case 5:
+                return values.get(0) + delimiter + values.get(1) + delimiter + values.get(2) + delimiter
+                    + values.get(3) + delimiter + values.get(4);
+            case 6:
+                return values.get(0) + delimiter + values.get(1) + delimiter + values.get(2) + delimiter
+                    + values.get(3) + delimiter + values.get(4) + delimiter + values.get(5);
+            case 7:
+                return values.get(0) + delimiter + values.get(1) + delimiter + values.get(2) + delimiter
+                    + values.get(3) + delimiter + values.get(4) + delimiter + values.get(5) + delimiter + values.get(6);
+            case 8:
+                return values.get(0) + delimiter + values.get(1) + delimiter + values.get(2) + delimiter
+                    + values.get(3) + delimiter + values.get(4) + delimiter + values.get(5) + delimiter + values.get(6)
+                    + delimiter + values.get(7);
+            case 9:
+                return values.get(0) + delimiter + values.get(1) + delimiter + values.get(2) + delimiter
+                    + values.get(3) + delimiter + values.get(4) + delimiter + values.get(5) + delimiter + values.get(6)
+                    + delimiter + values.get(7) + delimiter + values.get(8);
+            case 10:
+                return values.get(0) + delimiter + values.get(1) + delimiter + values.get(2) + delimiter
+                    + values.get(3) + delimiter + values.get(4) + delimiter + values.get(5) + delimiter + values.get(6)
+                    + delimiter + values.get(7) + delimiter + values.get(8) + delimiter + values.get(9);
+            default:
+                return String.join(delimiter, values);
+        }
+    }
+
+    /**
+     * Converts a byte array into a hex string.
+     * <p>
+     * If {@code uppercase} is true {@code ABCDEF} will be used, if false {@code abcdef} will be used. No matter the
+     * value of {@code uppercase} characters {@code 0123456789} remain consistent.
+     * <p>
+     * If {@code bytes} is null, null will be returned. If {@code bytes} was an empty array an empty string is
+     * returned.
+     *
+     * @param bytes The byte array to convert into a hex string.
+     * @param uppercase Whether {@code abcdef} or {@code ABCDEF} are used for the hex string.
+     * @return A hex string representing the {@code bytes} that were passed, or null if {@code bytes} were null.
+     */
+    public static String bytesToHexString(byte[] bytes, boolean uppercase) {
```

Comment:
None of your calls appear to pass `true` to the `uppercase` param. Is this necessary at this stage, or could you introduce an overload later on to avoid all the new `false` args being added.

Reply:
I'll remove the boolean and only support lowercase for now. There are some duplicates of this code but they all appear to be using lowercase too